### PR TITLE
issue: 825799 Add locking to setsockopt() TCP

### PR DESCRIPTION
**Pull request by OphirMunk**
Add locking to protect against simultaneous pcb changes by user applications and internal thread

Signed-off-by: Ophir Munk <ophirmu@mellanox.com>

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/209/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/222/ for details (Mellanox internal link).

File: src/vma/sock/sockinfo_tcp.cpp [hunk 1]
```diff
@@ -3101,36 +3101,44 @@ int sockinfo_tcp::setsockopt(int __level, int __optname,
 	if (__level == SOL_SOCKET) {
 		switch(__optname) {
 		case SO_REUSEADDR:
+			lock_tcp_con();
```
Comment (rosenbaumalex) [inline src/vma/sock/sockinfo_tcp.cpp]: @OphirMunk I recommend to locate the lock() just before the update of m_pcb. After the reading the optval and logging. no point logging inside the lock.
Same comment for all the rest of the changes.

**Comment by rosenbaumalex** [comment]
@OphirMunk need to rebase and then Jenkins will also retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/378/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/360/ for details (Mellanox internal link).

**Comment by rosenbaumalex** [comment]
:+1:

**Pull request merged**